tests/frontend/specs/chat.js

```diff
@@ -2,134 +2,86 @@ describe("Chat messages and UI", function(){
 
-  it("makes sure that an empty message can't be sent", function(done) {
-    var inner$ = helper.padInner$;
-    var chrome$ = helper.padChrome$;
+    // <p data-authorid="a.qjkwNs4z0pPROphS"
+    //   class="author-a-qjkwz78zs4z122z0pz80zz82zz79zphz83z">
+    //   <b>unnamed:</b>
+    //   <span class="time author-a-qjkwz78zs4z122z0pz80zz82zz79zphz83z">12:38
+    //   </span> JohnMcLear
+    // </p>
+    let username = helper.chatTextParagraphs().children("b").text();
+    let time = helper.chatTextParagraphs().children(".time").text();
+
+    expect(helper.chatTextParagraphs().text()).to.be(`${username}${time} ${chatValue}`);
 
-    //click on the chat button to make chat visible
-    var $chatButton = chrome$("#chaticon");
-    $chatButton.click();
-    var $chatInput = chrome$("#chatinput");
-    $chatInput.sendkeys('{enter}'); // simulate a keypress of enter (to send an empty message)
-    $chatInput.sendkeys('mluto'); // simulate a keypress of typing mluto
-    $chatInput.sendkeys('{enter}'); // simulate a keypress of enter (to send 'mluto')
-
-    //check if chat shows up
-    helper.waitFor(function(){
-      return chrome$("#chattext").children("p").length !== 0; // wait until the chat message shows up
-    }).done(function(){
-      // check that the empty message is not there
-      expect(chrome$("#chattext").children("p").length).to.be(1);
-      // check that the received message is not the empty one
-      var $firstChatMessage = chrome$("#chattext").children("p");
-      var containsMessage = $firstChatMessage.text().indexOf("mluto") !== -1;
-      expect(containsMessage).to.be(true);
-      done();
-    });
+    await helper.hideChat();
   });
 
-  it("makes chat stick to right side of the screen", function(done) {
-    var inner$ = helper.padInner$;
-    var chrome$ = helper.padChrome$;
+  it("makes sure that an empty message can't be sent", async function() {
+    var chatValue = "mluto";
 
-    //click on the settings button to make settings visible
-    var $settingsButton = chrome$(".buttonicon-settings");
-    $settingsButton.click();
+    await helper.showChat();
 
-    //get the chat selector
-    var $stickychatCheckbox = chrome$("#options-stickychat");
+    let failed = false;
+    // simulate a keypress of enter (to send an empty message)
+    await helper.sendChatMessage("{enter}")
+      .catch(function(){failed = true});
 
-    //select chat always on screen
-    if (!$stickychatCheckbox.is(':checked')) {
-      $stickychatCheckbox.click();
-    }
+    expect(failed).to.be(true);
 
-    // due to animation, we need to make some timeout...
-    setTimeout(function() {
-      //check if chat changed to get the stickychat Class
-      var $chatbox = chrome$("#chatbox");
-      var hasStickyChatClass = $chatbox.hasClass("stickyChat");
-      expect(hasStickyChatClass).to.be(true);
+    await helper.sendChatMessage(`${chatValue}{enter}`); // simulate a keypress of typing mluto and enter (to send 'mluto')
```

Comment:
This and later lines are now unnecessary due to the way you restructured the test.